core/src/core/block.rs

```diff
@@ -432,10 +432,9 @@ pub struct UntrustedBlockHeader(BlockHeader);
 impl Readable for UntrustedBlockHeader {
 	fn read<R: Reader>(reader: &mut R) -> Result<UntrustedBlockHeader, ser::Error> {
 		let header = read_block_header(reader)?;
-		if header.timestamp
-			> Utc::now() + Duration::seconds(12 * (consensus::BLOCK_TIME_SEC as i64))
-		{
-			// refuse blocks more than 12 blocks intervals in future (as in bitcoin)
+		if header.timestamp > Utc::now() + Duration::seconds(5 * 60) {
```

Comment:
@tromp this is what you were asking in keybase about?

We do not have access to config directly here. But we could potentially expose it via `lazy_static!` and `thread_local!` similar to what we do for `CHAIN_TYPE` and `NRD_FEATURE_ENABLED`.

That way we could have something like `global::get_future_time_limit()` following the same pattern as `global::get_chain_type()`.

The reason for introducing the `thread_local!` complexity is to make this configurable in tests. Otherwise we have a single global config value that ends up causing havoc across multiple tests.

Reply:
I'm not sure this is something we would want to test anytime soon.
Is implementation much simpler if we forego testing different settings?

Reply:
Not really. We're better off just following the existing pattern I think.
Or just leaving it as hardcoded for now. Maybe just move the constant to `global.rs` and reference it from there.

Reply:
Following existing pattern now.